docker/build_docker_windows.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Script to build docker image for the project."""
+
+import os
+import shutil
+import subprocess
+import urllib.request
+
+# First, get all environment variables that start with AWP_ROOT
+awp_root = {}
+for env_key, env_val in os.environ.items():
+    if env_key.startswith("AWP_ROOT"):
+        # There is an Ansys installation... Check that the version is at
+        # least 2023R2. Environment variables are in the form
+        # AWP_ROOT232=/path/to/2023R2
+        #
+        # Get the version number
+        version = env_key.split("AWP_ROOT")[1]
+        if version < "232":
+            # This version is too old, so we will ignore it
+            continue
+        else:
+            # This version is new enough, so we will add it to the list
+            awp_root[version] = env_val
+
+if len(awp_root) == 0:
+    # There are no Ansys installations
+    print("XXXXXXX No Ansys installations found.. exiting process. XXXXXXX")
+    exit(0)
```

Comment:
If it won't work because the version they have is too old, you should let the user know.